src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,176 @@
+            }
+            String details = inputArr[1];
+            String[] detailsArr = details.split(" /by ", 2);
+            if (detailsArr.length <= 1) {
+                throw new DukeException("Do you forget to add a deadline for the task?");
+            }
+            String des1 = detailsArr[0];
+            String by = detailsArr[1];
+            LocalDate byDate;
+            try {
+                byDate = LocalDate.parse(by);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("There is something wrong with your date! format: yyyy-mm-dd");
+            }
+            return new DeadlineCommand(des1, byDate);
+
+        case "event":
+            if (inputArr.length == 1) {
+                throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+            }
+            String detail = inputArr[1];
+            String[] detailArr = detail.split(" /from ", 2);
+            if (detailArr.length <= 1) {
+                throw new DukeException("Ouuuu, I think you miss some information, try again!");
+            }
+            String des2 = detailArr[0];
+            String time = detailArr[1];
+            String[] timeArr = time.split(" /to ", 2);
+            if (timeArr.length <= 1) {
+                throw new DukeException("Ouuuu, I think you miss some information, try again!");
+            }
+            String start = timeArr[0];
+            String end = timeArr[1];
+            LocalDate startDate;
+            LocalDate endDate;
+            try {
+                startDate = LocalDate.parse(start);
+                endDate = LocalDate.parse(end);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("There is something wrong with your date! format: yyyy-mm-dd");
+            }
+
+            return new EventCommand(des2, startDate, endDate);
+
+        case "mark":
+            if (inputArr.length == 1) {
+                throw new DukeException("OOPS!!! The description of a mark cannot be empty.");
+            }
+            if (input.split(" ").length > 2) {
+                throw new DukeException("Invalid mark command ?_? " +
+                        "this command should follow by only ONE INTEGER");
+            }
+            int i;
+            try {
+                i = Integer.parseInt(inputArr[1]);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please type in INTEGER after this command ^v^");
+            }
+
+            return new MarkCommand(i - 1);
+
+        case "unmark":
+            if (inputArr.length == 1) {
+                throw new DukeException("OOPS!!! The description of a unmark cannot be empty.");
+            }
+            if (input.split(" ").length > 2) {
+                throw new DukeException("Invalid unmark command ?_? " +
+                        "this command should follow by only ONE INTEGER");
+            }
+            int j;
```

Comment:
Try to be more consistent in variable naming, you used `i` for the mark case and `j` here